src/scores/emerging/emerging_impl.py

```diff
@@ -0,0 +1,478 @@
+):
+    """
+    Checks that the risk matrix score inputs are suitable.
+    """
+    if severity_dim not in fcst.dims:
+        raise ValueError("`severity_dim` must be a dimension of `fcst`")
+    if severity_dim not in obs.dims:
+        raise ValueError("`severity_dim` must be a dimension of `obs`")
+    if severity_dim not in decision_weights.dims:
+        raise ValueError("`severity_dim` must be a dimension of `decision_weights`")
+    if weights is not None and severity_dim in weights.dims:
+        raise ValueError("`severity_dim` must not be a dimension of `weights`")
+    if prob_threshold_dim in fcst.dims:
+        raise ValueError("`prob_threshold_dim` must not be a dimension of `fcst`")
+    if prob_threshold_dim in obs.dims:
+        raise ValueError("`prob_threshold_dim` must not be a dimension of `obs`")
+    if prob_threshold_dim not in decision_weights.dims:
+        raise ValueError("`prob_threshold_dim` must be a dimension of `decision_weights`")
+    if weights is not None and prob_threshold_dim in weights.dims:
+        raise ValueError("`prob_threshold_dim` must not be a dimension of `weights`")
+    if len(decision_weights.dims) != 2:
+        raise ValueError("`decision_weights` must have exactly 2 dimensions: `severity_dim` and `prob_threshold_dim`")
+
+    if isinstance(fcst, xr.Dataset):
+        fcst = fcst.to_array()
+
+    if (fcst.max() > 1) or (fcst.min() < 0):
+        raise ValueError("values in `fcst` must lie in the closed interval `[0, 1]`")
+
+    if isinstance(obs, xr.Dataset):
+        obs = obs.to_array()
+
+    disallowed_obs_values = np.unique(obs.where(obs != 1).where(obs != 0).values)
+    disallowed_obs_values = disallowed_obs_values[~np.isnan(disallowed_obs_values)]
+    if len(disallowed_obs_values) > 0:
+        raise ValueError("values in `obs` can only be 0, 1 or nan")
+
+    if (decision_weights[prob_threshold_dim].min() <= 0) or (decision_weights[prob_threshold_dim].max() >= 1):
+        raise ValueError("`prob_threshold_dim` coordinates must be strictly between 0 and 1")
+
+    if not np.array_equal(np.sort(decision_weights[severity_dim].values), np.sort(fcst[severity_dim].values)):
+        raise ValueError("`severity_dim` coordinates do not match in `decision_weights` and `fcst`")
+    if not np.array_equal(np.sort(decision_weights[severity_dim].values), np.sort(obs[severity_dim].values)):
+        raise ValueError("`severity_dim` coordinates do not match in `decision_weights` and `obs`")
+
+    if threshold_assignment not in ["upper", "lower"]:
+        raise ValueError(""" `threshold_assignment` must be either \"upper\" or \"lower\" """)
+
+
+def _risk_matrix_score(
+    fcst: XarrayLike,
+    obs: XarrayLike,
+    decision_weights: xr.DataArray,
+    severity_dim: str,
+    prob_threshold_dim: str,
+    threshold_assignment: Optional[str] = "lower",
+) -> xr.DataArray:
+    """
+    Calculates the risk matrix score MS of Taggart and Wilke (2024).
```

Comment:
"MS" is the abbreviation given to "Matrix Score" in the paper. In contrast to the warning score (WS) in the paper.

I'll leave it as is now, but it might need updating prior to the paper being submitted. We are thinking of using the abbreviation RMS instead.

Reply:
Given "MS" is an abbreviation, and it is the first time the abbreviation is being mentioned, I agree that it would be clearer to include it here in brackets - i.e. "(MS)".

Similarly, if down the road it is changed to RMS, I would suggest including it in brackets as "(RMS)" when it is first introduced as the abbreviation.

Reply:
OK, I'll go along with the suggestion. I was using mathematics paper writing conventions, and in this case one doesn't require parentheses if adhering to those conventions. This context is different I suppose.